hyppo/independence/mirf.py

```diff
@@ -37,6 +37,38 @@ def perm_stat(clf, x, z, y, random_state=None):
     return perm_stat
 
 
+def pos_diff(observe_pos, perm_pos, limit):
+    total_pos = np.random.shuffle(np.concatenate((observe_pos, perm_pos)))
+
+    half_ind = len(total_pos) * 0.5
+    half_pos = total_pos[:half_ind]
+    end_pos = total_pos[half_ind:]
+
+    half_pos_final = forest_pos(half_pos)
+    half_stat = roc_auc_score(half_pos_final[:, 1], half_pos_final[:, 2], max_fpr=limit)
+
+    end_pos_final = forest_pos(end_pos)
+    end_stat = roc_auc_score(end_pos_final[:, 1], end_pos_final[:, 2], max_fpr=limit)
+
+    return abs(half_stat - end_stat)
+
+
+def forest_pos(posterior):
+    posterior_matrix = np.array(posterior).reshape(-1, 3)
```

Comment:
It's not clear to me what this is doing wrt paper. Why is this split into two different indices? Perhaps we can document it explicitly in docstring?

Reply:
Here I combine the tree posteriors, make a shuffle & split them into two lists. Then I combine the tree posteriors to calculate the partial AUC.

Reply:
Why are there 3 columns necessarily tho?

Is it `(n_samples, n_outputs)`?

Reply:
no, because each posterior matrix must have the sample index, sample label, sample predictions. The indices are needed to combine posteriors from trees as each tree are trained and tested on different splits of samples.

This is just how we decided to calculate the statistic, not related to the paper.